### PR TITLE
[GUILD] - Convert `Select` to TS

### PR DESCRIPTION
**Pull request by qubis741**
### Description

This PR converts `Select` component to typescript.

### Manual check

- Check storybook and controls if they work

### CODE REVIEW

File: src/components/select/Select.tsx [hunk 1]
```diff
@@ -32,10 +103,17 @@ const DropdownIndicator = ({ selectProps: { inverse } }) => {
     </Icon>
   );
 };
-
-const ClearIndicator = ({ innerProps, selectProps: { inverse } }) => {
+//
+const ClearIndicator = <IsMulti extends boolean>(clearIndicatorProps: ClearIndicatorProps<OptionType, IsMulti>) => {
+  // @ts-ignore
+  const inverse = clearIndicatorProps.selectProps.inverse;
```
Comment (lorgan3) [inline src/components/select/Select.tsx:109]: Does this actually work? I don't see this `inverse` being passed to the react-select component anywhere 🤔

File: src/components/select/Select.tsx [hunk 2]
```diff
@@ -46,26 +124,43 @@ selectOverlayNode.setAttribute('data-teamleader-ui', 'select-overlay');
...
+    menuWidth,
+    width = '100%',
+    truncateOptionText,
+    options,
+    ...otherProps
+  }: SelectProps<Option, IsMulti, IsClearable>,
+  ref: ForwardedRef<SelectType<Option, IsMulti>>,
+) {
+  useEffect(() => {
+    // @ts-ignore implicit type of any, nothing to do here about it
```
Comment (lorgan3) [inline src/components/select/Select.tsx:147]: You can declare what `this` is like this: https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function

First argument of your `Select` function can be `this: GenericComponent<SelectProps>`

File: src/components/select/Select.tsx [hunk 3]
```diff
@@ -345,101 +423,67 @@ class Select extends PureComponent {
...
 
-/** @type {React.ComponentType<any>} */
-const ForwardedSelect = forwardRef((props, ref) => <Select {...props} forwardedRef={ref} />);
+  return (
+    <Box className={wrapperClassnames} {...boxProps}>
+      <Element
+        ref={ref}
+        className={cx(uiUtilities['reset-font-smoothing'], theme['select'])}
+        components={{
+          // @ts-ignore just some type generic magic
```
Comment (lorgan3) [inline src/components/select/Select.tsx:461]: Making `Option` generic seems to make the error go away 🤔 
```
const ClearIndicator = <IsMulti extends boolean, Option extends OptionType>(
  clearIndicatorProps: ClearIndicatorProps<Option, IsMulti>,
) => {
```

File: src/components/select/select.stories.tsx [hunk 4]
```diff
@@ -15,18 +36,24 @@ const CustomOption = ({ children, data, innerProps, isFocused, isSelected, isDis
   const textStyles = {
     color: isDisabled ? '#c0c0c4' : isSelected && !isFocused ? '#fff' : '#344b63',
   };
-
+  // @ts-ignore
```
Comment (lorgan3) [inline src/components/select/select.stories.tsx:39]: This doesn't seem to be necessary

File: src/components/select/types.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export type Value = any;
+
+export type Option<V = Value> = {
```
Comment (lorgan3) [inline src/components/select/types.ts:3]: Interface please 😅

**Review by lorgan3 — COMMENTED** [review]
Enabling `clearable` doesn't seem to work anymore (Might also be a storybook issue)

**Review by lorgan3 — DISMISSED** [review]

**Review by driesd — APPROVED** [review]

**Review by eniskraasniqi1 — DISMISSED** [review]

**Comment by eniskraasniqi1** [comment]
Checking ⏳

**Review by lorgan3 — APPROVED** [review]

**Review by eniskraasniqi1 — APPROVED** [review]
Checked ✅ . Also checked differences with production storybook, all the same 👌

**Review by driesd — APPROVED** [review]

**Pull request merged**